library/core/src/convert/mod.rs

```diff
@@ -532,7 +532,8 @@ where
 
 // From implies Into
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T, U> Into<U> for T
+#[rustc_const_unstable(feature = "const_identity_convert", issue = "none")]
+impl<T, U> const Into<U> for T
```

Comment:
With [comment from #86808](https://github.com/rust-lang/rust/pull/86808#issuecomment-892280144) in mind would you prefer if I removed this and saved it for another PR?

Reply:
yea, I think it is better to leave this until we have sufficient tests around this

Reply:
Btw, should this have its own feature gate since this is not necessarily an identity conversion?

Reply:
Then just make the feature gate const_convert.

Reply:
As in, replace all `const_identity_convert` with `const_convert`?

Reply:
Yeah, that would be a more suitable name for a feature covering both impls